__init__.py

```diff
@@ -12,314 +12,420 @@
+    def handle_status_timer(self, message):
         intent = message.data
-        amt_of_timer = len(self.active_timers)
-        if amt_of_timer == 0:
-            self.speak("Cannot find any active timers")
-        elif amt_of_timer == 1:
-            timer_name = self.active_timers[0]
-            self.speak_timer_status(timer_name)
-        elif amt_of_timer > 1:
-            self.multiple_timer_status(intent, amt_of_timer)
-
-    def multiple_timer_status(self, intent, amt_of_timer):
+        if not self.active_timers:
+            self.speak_dialog("no.active.timer")
+        elif len(self.active_timers) == 1:
+            self._speak_timer_status(self.active_timers[0]["name"])
+        else:
+            self._multiple_timer_status(intent)
+
+    def _multiple_timer_status(self, intent):
         """ determines which timer to speak the status of
 
             Args:
                 intent (dict): data from Message object
         """
         if 'duration' not in intent.keys():
-            self.should_converse = True
-            # let converse knows how to handle utterances
-            self.intent_context = 'status.timer.intent'
-            # when setting expect_respose = True, let's mycroft activate
-            # listening mode right after speak
-            mult_str = "You have {} active timers. ".format(amt_of_timer)
-            for i in range(len(self.active_timers)):
-                mult_str += "the {} timer, ".format(self.active_timers[i])
-                if (i + 2) == len(self.active_timers):
-                    mult_str += "and "
-                if ((i + 1) == len(self.active_timers)):
-                    mult_str += "."
-            mult_str += "Which one were you referring to?"
-            self.speak(mult_str, expect_response=True)
+            which = self.get_response('ask.which.timer',
+                                      data={"count": len(self.active_timers)})
+            if not which:
+                return
+
+            num = extractnumber(which)
+            if num and num-1 < len(self.active_timers):
+                timer = self.active_timers[int(num-1)]
+            else:
+                timer = None
+                best = 0.5  # minimum threshold for a match
+                for t in self.active_timers:
+                    if fuzzy_match(which, t["name"]) > best:
+                        best = fuzzy_match(which, t["name"])
+                        timer = t
+            if timer:
+                self._speak_timer_status(timer["name"])
         else:
             duration = intent['duration']
             timer_name = duration
-            self.speak_timer_status(timer_name)
+            self._speak_timer_status(timer_name)
+
+    @intent_handler(IntentBuilder("").require("Mute").require("Timer"))
+    def handle_mute_timer(self, message):
+        self.mute = True
 
+    @intent_handler(IntentBuilder("").require("Cancel").require("Timer").optionally("All"))
```

Comment:
cancel timer doesn't work by voice. The intent may not be registering?

Reply:
Also if multiple timers are set, and if you hit the top button, Mycroft will ask N times consecutively if you'd like to cancel the timer. N is the numbers of timers set.